docs/ics/ics-xxx-signed-messages.md

```diff
@@ -0,0 +1,184 @@
+to the Cosmos chain identifier. The user-agent should **refuse** signing if the
+`@chain_id` field does not match the currently active chain! The `@type` field
+must equal the constant `"message"`. The `@type` field corresponds to the type of 
+structure the user will be signing in an application. For now, a user is only
+allowed to sign bytes of valid ASCII text ([see here](https://github.com/tendermint/tendermint/blob/master/libs/common/string.go#L61-L74)).
+However, this will change and evolve to support additional application-specific
+structures that are human-readable and machine-verifiable ([see Future Adaptations](#futureadaptations)).
+
+Thus, we can have a canonical JSON structure for signing Cosmos messages using
+the [JSON schema](http://json-schema.org/) specification as such:
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "$id": "cosmos/signing/typeData/schema",
+  "title": "The Cosmos signed message typed data schema.",
+  "type": "object",
+  "properties": {
+    "@chain_id": {
+      "type": "string",
+      "description": "The corresponding Cosmos chain identifier.",
+      "minLength": 1
+    },
+    "@type": {
+      "type": "string",
+      "description": "The message type. It must be 'message'.",
+      "enum": [
+        "message"
+      ]
+    },
+    "text": {
+      "type": "string",
+      "description": "The valid ASCII text to sign.",
+      "pattern": "^[\\x20-\\x7E]+$",
+      "minLength": 1
+    }
+  },
+  "required": [
+    "@chain_id",
+    "@type",
+    "text"
+  ]
+}
+```
+
+e.g.
+```json
+{
+  "@chain_id": "1",
+  "@type": "message",
+  "text": "Hello, you can identify me as XYZ on keybase."
+}
+```
+
+## Future Adaptations
+
+As applications can vary greatly in domain, it will be vital to support both
+domain separation and human-readable and machine-verifiable structures.
+
+Domain separation will allow for application developers to prevent collisions of
+otherwise identical structures. It should be designed to be unique per application
+use and should directly be used in the signature encoding itself.
+
+Human-readable and machine-verifiable structures will allow end users to sign
+more complex structures, apart from just string messages, and still be able to
+know exactly what they are signing (opposed to signing a bunch of arbitrary bytes).
+
+Thus, in the future, the Cosmos signing message specification will be expected
+to expand upon it's canonical JSON structure to include such functionality.
+
+
+## API
+
+Application developers and designers should formalize a standard set of APIs that
+adhere to the following specification:
+
+<hr>
+
+**cosmosSignBytes**
+
+Params:
+* `data`: the Cosmos signed message canonical JSON structure
+* `address`: 20 byte account address to sign data with
+
+Returns:
+* `signature`: the Cosmos signature derived using signing algorithm `S`
+
+<hr>
+
+**cosmosSignBytesPass**
```

Comment:
Maybe expand to `cosmosSignBytesPassword`?

Reply:
Yeah I can see how `pass` may seem a bit..."off". I'll update 👍